addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/discovery/BigAssFanDiscoveryService.java

```diff
@@ -0,0 +1,270 @@
+        logger.info("BigAssFan discovery service is running");
+
+        try {
+            discoveryListener = new DiscoveryListener();
+        } catch (SocketException se) {
+            logger.warn("Got Socket exception creating multicast socket: {}", se.getMessage());
+            return;
+        } catch (IOException ioe) {
+            logger.warn("Got IO exception creating multicast socket: {}", ioe.getMessage());
+            return;
+        }
+
+        logger.debug("Waiting for discovery messages");
+        while (!terminate) {
+            try {
+                // Wait for a discovery message
+                BigAssFanDevice device = discoveryListener.waitForMessage();
+                processMessage(device);
+            } catch (IOException ioe) {
+                logger.warn("Got IO exception waiting for message: {}", ioe.getMessage());
+                break;
+            } catch (Exception e) {
+                logger.warn("Got exception waiting for message: {}", e.getMessage());
+                break;
+            }
+        }
+        discoveryListener.shutdown();
+        logger.debug("DiscoveryListener job is exiting");
+    }
+
+    private void processMessage(BigAssFanDevice device) {
+        if (device == null) {
+            return;
+        }
+        // XXX Uncomment to test discovery
+        // device.setDiscoveryMessage("(Master Bedroom Wall Controller;DEVICE;ID;20:F8:5E:DB:98:37;SWITCH,SENSEME)");
+        // device.setDiscoveryMessage("(Master Bedroom Fan;DEVICE;ID;20:F8:5E:DA:A5:6A;FAN,LSERIES)");
+        // device.setIpAddress("192.168.1.99");
+        // device.setDiscoveryMessage("(Porch Wall Controller;DEVICE;ID;20:F8:5E:DB:98:A7;SWITCH,SENSEME)");
+        // device.setDiscoveryMessage("(Porch Fan;DEVICE;ID;20:F8:5E:DA:A5:AA;FAN,LSERIES)");
+        // device.setIpAddress("192.168.1.98");
+
+        Pattern pattern = Pattern.compile("[(](.*);DEVICE;ID;(.*);(.*)[)]");
+        Matcher matcher = pattern.matcher(device.getDiscoveryMessage());
+        if (matcher.find()) {
+            logger.debug("Match: grp1={}, grp2={}, grp(3)={}", matcher.group(1), matcher.group(2), matcher.group(3));
+
+            // Extract needed information from the discovery message
+            device.setLabel(matcher.group(1));
+            device.setMacAddress(matcher.group(2));
+            String[] modelParts = matcher.group(3).split(",");
+            if (modelParts.length != 2) {
+                logger.info("Unable to extract device type from discovery message");
+                return;
+            }
+            device.setType(modelParts[0]);
+            device.setModel(modelParts[1]);
+
+            deviceDiscovered(device);
+        }
+    }
+
+    private synchronized void deviceDiscovered(BigAssFanDevice device) {
+        logger.info("Device discovered: {}", device);
+
+        if (device.isSwitch()) {
+            logger.debug("Switches (controllers) are not handled currently");
```

Comment:
I understand your point.  But, I don't think normal end-users would be interested in this.  I haven't found anything that a user can actually do with/to the wall controllers.

I would suggest the following:
- leave switches as debug, unless and until there's something meaningful that a user can do with a wall controller, and
- make the unknown device message info, because I think the user will want to know about that.